# Analysis and prediction of Milk Price in Ireland

# Business Goal
Predict the Milk Price In Ireland, two non EU countries(New-Zealand, India) and an EU country(France). Also compare the Milk prices with Ireland as baseline with other countries in the world

# Data Preparation
Provide references for the data collected

# Milk Price Prediction
The goal of this project is to predict the milk price in Ireland based on the historical data and the predictor variables


# Table of Contents

# Section 1: Ireland Milk price analysis and prediction

# Section 2: New Zealand Milk price analysis and prediction

# Section 3: India Milk price analysis and prediction

# Section 4: France Milk Price analysis and Prediction

# Section 5: Comparison of Milk prices with Ireland and World

# Section 6: Statistical Analysis

# References
All the literature reviewed as part of this project is given in the reference section as Harvard style referencing

# Section 1: Ireland Milk price analysis and prediction

Phase 1 : Data understanding and exploration

Let's first have a look at the dataset and understand the size, attribute names etc

In [1]:
# Import the Required Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from warnings import filterwarnings
filterwarnings('ignore')

In [2]:
#read all the milk or dairy related feature datasets of Ireland collected from various sources
milk_sales_Ireland_monthly                 = pd.read_csv('All_milk_sales_Ireland_monthly.csv')            # all milk sales Ireland
calf_birth_beef_bull_female_monthly        = pd.read_csv('Calf_birth_Beef_Bull_Female_monthly.csv')       # calf birth beef bull female
calf_birth_beef_bull_male_monthly          = pd.read_csv('Calf_birth_Beef_Bull_Male_monthly.csv')         # calf birth beef bull male
calf_birth_dairy_bull_female_monthly       = pd.read_csv('Calf_birth_Dairy_Bull_Female_monthly.csv')      # calf birth dairy bull female
calf_birth_dairy_bull_male_monthly         = pd.read_csv('Calf_birth_Dairy_Bull_Male_monthly.csv')        # calf birth dairy bull male
cattle_fattening_feed_price_monthly        = pd.read_csv('Cattle_fattening_feed_price_monthly.csv')       # cattle fattening feed
cattle_fattening_nuts_cubes_price_monthly  = pd.read_csv('Cattle_fattening_nuts_cubes_price_monthly.csv') # cattle fattening nuts cubes
consumer_price_fullfat_milk2L_monthly      = pd.read_csv('Consumer_price_fullfat_milk_2L_euro.csv')       # consumer price fullfat milk
consumer_price_lowfat_milk1L_monthly       = pd.read_csv('Consumer_price_lowfat_milk_1L_euro.csv')        # consumer price lowfat milk
cow_milk_intake_pasteury_creamery_monthly  = pd.read_csv('Cows_milk_intake_domestic_creameris_pasteuries_monthly.csv') # milk intake creamery pasteury
fertilizer_calciumammonium_nitrate_monthly = pd.read_csv('Fertilizer_calciumammonium_nitrate_price_monthly.csv') # fertilizer calciumammonium nitrate price
fertilizer_granularsuperphosphate_monthly  = pd.read_csv('Fertilizer_Granular_Superphosphate_price_monthly.csv') # fertilizer superphosphate price
fertilizer_potash_muriate_monthly          = pd.read_csv('Fertilizer_muriate_potash_price_monthly.csv') # fertilizer potash muriate price
fertilizer_urea_monthly                    = pd.read_csv('Fertilizer_Urea_price_monthly.csv') # fertilizer urea price
heifer_200_249kg_price_monthly             = pd.read_csv('Heifers_200_249KG_price_monthly.csv') # heifer 200 249kg price
heifer_250_299kg_price_monthly             = pd.read_csv('Heifers_250_299KG_price_monthly.csv') # heifer 250 299kg price
heifer_300_349kg_price_monthly             = pd.read_csv('Heifers_300_349KG_price_monthly.csv') # heifer 300 349kg price
heifer_350_399kg_price_monthly             = pd.read_csv('Heifers_350_399KG_price_monthly.csv') # heifer 350 399kg price
heifer_400_449kg_price_monthly             = pd.read_csv('Heifers_400_449KG_price_monthly.csv') # heifer 400 449kg price
consumer_price_index_Ireland_monthly       = pd.read_excel('Ireland_Consumer_Price_Index_monthly.xlsx') # Ireland consumer price
producer_price_index_Ireland_monthly       = pd.read_excel('Ireland_Producer_Price_Index_monthly.xlsx') # Ireland producer price
maize_meal_price_Ireland_monthly           = pd.read_csv('Maize_meal_price_Ireland_monthly.csv') # Maize meal price
producer_price_actualprotein_milk_monthly  = pd.read_csv('Manufacture_price_actualfatprotein_milk_monthly.csv') # Manufacture price milk
milk_sales_skimmed_semiskimmed_Ireland     = pd.read_csv('Milk_sales_skimmed_semiskimmed_Ireland_monthly.csv') # Milk sales semiskimmed and skimmed
milk_sales_whole_milk_Ireland              = pd.read_csv('Milk_sales_whole_milk_Ireland_monthly.csv') # Milk sales whole milk
butter_production_monthly                  = pd.read_csv('Production_Butter_monthly.csv') # butter production
cheese_production_monthly                  = pd.read_csv('Production_Cheese_monthly.csv') # cheese production
skimmed_milk_production_monthly            = pd.read_csv('Production_skimmed_milk_monthly.csv') # skimmed milk production
transport_consumer_price_index_monthly     = pd.read_csv('Transport_Consumer_PriceIndex_monthly.csv') # transport consumer priceindex
wheat_feed_price_Ireland_monthly           = pd.read_csv('Wheat_feed_price_Ireland_monthly.csv') # wheat feed price
diesel_price_monthly                       = pd.read_csv('Diesel_price_monthly.csv') # diesel price index
petrol_price_monthly                       = pd.read_csv('Petrol_unleaded_price_monthly.csv') # unleaded petrol price index

Analaysis of all the observations by displaying the records, shape and datatypes

In [10]:
#check first 5 records
milk_sales_Ireland_monthly.head()

Statistic    Month Type of Milk  \
0  Milk Sales (Dairy) for Human Consumption  1980M01     All Milk   
1  Milk Sales (Dairy) for Human Consumption  1980M02     All Milk   
2  Milk Sales (Dairy) for Human Consumption  1980M03     All Milk   
3  Milk Sales (Dairy) for Human Consumption  1980M04     All Milk   
4  Milk Sales (Dairy) for Human Consumption  1980M05     All Milk   

             UNIT  VALUE  
0  Million Litres   38.2  
1  Million Litres   36.6  
2  Million Litres   38.4  
3  Million Litres   36.8  
4  Million Litres   38.8

In [4]:
#check last 5 records
milk_sales_Ireland_monthly.tail()

Statistic    Month Type of Milk  \
501  Milk Sales (Dairy) for Human Consumption  2021M10     All Milk   
502  Milk Sales (Dairy) for Human Consumption  2021M11     All Milk   
503  Milk Sales (Dairy) for Human Consumption  2021M12     All Milk   
504  Milk Sales (Dairy) for Human Consumption  2022M01     All Milk   
505  Milk Sales (Dairy) for Human Consumption  2022M02     All Milk   

               UNIT  VALUE  
501  Million Litres   41.9  
502  Million Litres   40.9  
503  Million Litres   38.9  
504  Million Litres   40.3  
505  Million Litres   36.1

In [5]:
#check for shape
milk_sales_Ireland_monthly.shape

(506, 5)

In [9]:
#check data type
milk_sales_Ireland_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Statistic     506 non-null    object 
 1   Month         506 non-null    object 
 2   Type of Milk  506 non-null    object 
 3   UNIT          506 non-null    object 
 4   VALUE         506 non-null    float64
dtypes: float64(1), object(4)
memory usage: 19.9+ KB


In [11]:
#check for missing or null values in the data
milk_sales_Ireland_monthly.isnull().sum()

Statistic       0
Month           0
Type of Milk    0
UNIT            0
VALUE           0
dtype: int64

Insights gained from above analysis for Ireland Milk sales data
1. Contains records from 1980 to 2022 with total 506 observations and 5 columns
2. No null values in the data
3. Month value is in object datatype and sales data is in float datatype
4. Milk unit is in million litres

In [12]:
#check first 5 records
calf_birth_beef_bull_female_monthly.head()

Month       UNIT  Beef Bull Female count
0  2015M01  Headcount                   45797
1  2015M02  Headcount                  103117
2  2015M03  Headcount                  159785
3  2015M04  Headcount                  140890
4  2015M05  Headcount                   86454

In [13]:
#check last 5 records
calf_birth_beef_bull_female_monthly.tail()

Month       UNIT  Beef Bull Female count
81  2021M10  Headcount                   20426
82  2021M11  Headcount                   17647
83  2021M12  Headcount                   14318
84  2022M01  Headcount                   49099
85  2022M02  Headcount                  122786

In [14]:
#check shape
calf_birth_beef_bull_female_monthly.shape

(86, 3)

In [16]:
#check data type
calf_birth_beef_bull_female_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86 entries, 0 to 85
Data columns (total 3 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Month                   86 non-null     object
 1   UNIT                    86 non-null     object
 2   Beef Bull Female count  86 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 2.1+ KB


In [15]:
#check for missing or null values in the data
calf_birth_beef_bull_female_monthly.isnull().sum()

Month                     0
UNIT                      0
Beef Bull Female count    0
dtype: int64

Insights gained from above analysis for Calf birth beef female bull data

Contains records from 2015 to 2022 with total 86 observations and 3 columns
No null values in the data
Month value is in object datatype and calf birth headcount is integer datatype
unit is in headcount or number

In [17]:
#check first 5 records
calf_birth_beef_bull_male_monthly.head()

Month       UNIT  Beef Bull Male count
0  2015M01  Headcount                 47039
1  2015M02  Headcount                108548
2  2015M03  Headcount                171057
3  2015M04  Headcount                151498
4  2015M05  Headcount                 93068

In [18]:
#check last 5 records
calf_birth_beef_bull_male_monthly.tail()

Month       UNIT  Beef Bull Male count
81  2021M10  Headcount                 21420
82  2021M11  Headcount                 18802
83  2021M12  Headcount                 15184
84  2022M01  Headcount                 50869
85  2022M02  Headcount                131514

In [19]:
#check shape
calf_birth_beef_bull_male_monthly.shape

(86, 3)

In [20]:
#check data type
calf_birth_beef_bull_male_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86 entries, 0 to 85
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Month                 86 non-null     object
 1   UNIT                  86 non-null     object
 2   Beef Bull Male count  86 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 2.1+ KB


In [21]:
#check for missing or null values in the data
calf_birth_beef_bull_male_monthly.isnull().sum()

Month                   0
UNIT                    0
Beef Bull Male count    0
dtype: int64

Insights gained from above analysis for Calf birth beef male bull data

1. Contains records from 2015 to 2022 with total 86 observations and 3 columns
2. No null values in the data 
3. Month value is in object datatype and calf birth headcount is integer datatype 
4. unit is in headcount or number

In [23]:
#check first 5 records
calf_birth_dairy_bull_female_monthly.head()

Month       UNIT  Diary Bull Female count
0  2015M01  Headcount                    62021
1  2015M02  Headcount                   159140
2  2015M03  Headcount                    72247
3  2015M04  Headcount                    29121
4  2015M05  Headcount                    12348

In [24]:
#check last 5 records
calf_birth_dairy_bull_female_monthly.tail()

Month       UNIT  Diary Bull Female count
81  2021M10  Headcount                    11304
82  2021M11  Headcount                     8169
83  2021M12  Headcount                     6100
84  2022M01  Headcount                    94700
85  2022M02  Headcount                   212620

In [25]:
#check shape
calf_birth_dairy_bull_female_monthly.shape

(86, 3)

In [26]:
#check data type
calf_birth_dairy_bull_female_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86 entries, 0 to 85
Data columns (total 3 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Month                    86 non-null     object
 1   UNIT                     86 non-null     object
 2   Diary Bull Female count  86 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 2.1+ KB


In [27]:
#check for missing or null values
calf_birth_dairy_bull_female_monthly.isnull().sum()

Month                      0
UNIT                       0
Diary Bull Female count    0
dtype: int64

Insights gained from above analysis for Calf birth dairy female bull data

1. Contains records from 2015 to 2022 with total 86 observations and 3 columns
2. No null values in the data
3. Month value is in object datatype and calf birth headcount is integer datatype
4. unit is in headcount or number

In [28]:
#check first 5 records
calf_birth_dairy_bull_male_monthly.head()

Month       UNIT  Diary Bull Male count
0  2015M01  Headcount                  53833
1  2015M02  Headcount                 160713
2  2015M03  Headcount                  77789
3  2015M04  Headcount                  30634
4  2015M05  Headcount                  13193

In [29]:
#check last 5 records
calf_birth_dairy_bull_male_monthly.tail()

Month       UNIT  Diary Bull Male count
81  2021M10  Headcount                   9766
82  2021M11  Headcount                   7422
83  2021M12  Headcount                   5280
84  2022M01  Headcount                  76064
85  2022M02  Headcount                 205844

In [30]:
#check shape
calf_birth_dairy_bull_male_monthly.shape

(86, 3)

In [31]:
#check data type
calf_birth_dairy_bull_male_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86 entries, 0 to 85
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Month                  86 non-null     object
 1   UNIT                   86 non-null     object
 2   Diary Bull Male count  86 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 2.1+ KB


In [32]:
#check null values or missing values
calf_birth_dairy_bull_male_monthly.isnull().sum()

Month                    0
UNIT                     0
Diary Bull Male count    0
dtype: int64

Insights gained from above analysis for Calf birth dairy male bull data

1. Contains records from 2015 to 2022 with total 86 observations and 3 columns
2. No null values in the data
3. Month value is in object datatype and calf birth headcount is integer datatype
4. unit is in headcount or number

In [33]:
#check for first 5 records
cattle_fattening_feed_price_monthly.head()

Statistic    Month                       Type of Feedstuff  \
0  Feed Stuff Price  2014M01  Cattle fattening meal (13-15% protein)   
1  Feed Stuff Price  2014M02  Cattle fattening meal (13-15% protein)   
2  Feed Stuff Price  2014M03  Cattle fattening meal (13-15% protein)   
3  Feed Stuff Price  2014M04  Cattle fattening meal (13-15% protein)   
4  Feed Stuff Price  2014M05  Cattle fattening meal (13-15% protein)   

             UNIT  VALUE  
0  Euro per Tonne    296  
1  Euro per Tonne    296  
2  Euro per Tonne    281  
3  Euro per Tonne    277  
4  Euro per Tonne    282

In [34]:
#check for last 5 records
cattle_fattening_feed_price_monthly.tail()

Statistic    Month                       Type of Feedstuff  \
93  Feed Stuff Price  2021M10  Cattle fattening meal (13-15% protein)   
94  Feed Stuff Price  2021M11  Cattle fattening meal (13-15% protein)   
95  Feed Stuff Price  2021M12  Cattle fattening meal (13-15% protein)   
96  Feed Stuff Price  2022M01  Cattle fattening meal (13-15% protein)   
97  Feed Stuff Price  2022M02  Cattle fattening meal (13-15% protein)   

              UNIT  VALUE  
93  Euro per Tonne    326  
94  Euro per Tonne    330  
95  Euro per Tonne    339  
96  Euro per Tonne    351  
97  Euro per Tonne    353

In [35]:
#check for shape
cattle_fattening_feed_price_monthly.shape

(98, 5)

In [36]:
#check the data type
cattle_fattening_feed_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 98 entries, 0 to 97
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Statistic          98 non-null     object
 1   Month              98 non-null     object
 2   Type of Feedstuff  98 non-null     object
 3   UNIT               98 non-null     object
 4   VALUE              98 non-null     int64 
dtypes: int64(1), object(4)
memory usage: 4.0+ KB


In [37]:
#check for missing values or null values
cattle_fattening_feed_price_monthly.isnull().sum()

Statistic            0
Month                0
Type of Feedstuff    0
UNIT                 0
VALUE                0
dtype: int64

Insights gained from above analysis for Cattle fattening feed data

1. Contains records from 2014 to 2022 with total 98 observations and 5 columns
2. No null values in the data
3. Month value is in object datatype and cattle fattening feed is integer datatype
4. unit is in euro per tonne

In [39]:
#check first 5 records
cattle_fattening_nuts_cubes_price_monthly.head()

Statistic    Month  \
0  Feed Stuff Price  2014M01   
1  Feed Stuff Price  2014M02   
2  Feed Stuff Price  2014M03   
3  Feed Stuff Price  2014M04   
4  Feed Stuff Price  2014M05   

                                  Type of Feedstuff            UNIT  VALUE  
0  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    294  
1  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    294  
2  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    280  
3  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    283  
4  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    289

In [40]:
#check last 5 records
cattle_fattening_nuts_cubes_price_monthly.tail()

Statistic    Month  \
93  Feed Stuff Price  2021M10   
94  Feed Stuff Price  2021M11   
95  Feed Stuff Price  2021M12   
96  Feed Stuff Price  2022M01   
97  Feed Stuff Price  2022M02   

                                   Type of Feedstuff            UNIT  VALUE  
93  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    329  
94  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    333  
95  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    345  
96  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    356  
97  Cattle fattening nuts and cubes (13-15% protein)  Euro per Tonne    363

In [41]:
#check shape
cattle_fattening_nuts_cubes_price_monthly.shape

(98, 5)

In [42]:
#check data type
cattle_fattening_nuts_cubes_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 98 entries, 0 to 97
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Statistic          98 non-null     object
 1   Month              98 non-null     object
 2   Type of Feedstuff  98 non-null     object
 3   UNIT               98 non-null     object
 4   VALUE              98 non-null     int64 
dtypes: int64(1), object(4)
memory usage: 4.0+ KB


In [43]:
#check for missing values or null
cattle_fattening_nuts_cubes_price_monthly.isnull().sum()

Statistic            0
Month                0
Type of Feedstuff    0
UNIT                 0
VALUE                0
dtype: int64

Insights gained from above analysis for Cattle fattening nuts cubes prices data

1. Contains records from 2014 to 2022 with total 98 observations and 5 columns
2. No null values in the data
3. Month value is in object datatype and cattle feed price is integer datatype
4. unit is in euro per tonne

In [45]:
#check first 5 records
consumer_price_fullfat_milk2L_monthly.head()

Statistic    Month              Consumer Item  UNIT  VALUE
0  National Average Price  2011M12  Full fat milk per 2 litre  Euro    NaN
1  National Average Price  2012M01  Full fat milk per 2 litre  Euro  1.778
2  National Average Price  2012M02  Full fat milk per 2 litre  Euro  1.775
3  National Average Price  2012M03  Full fat milk per 2 litre  Euro  1.774
4  National Average Price  2012M04  Full fat milk per 2 litre  Euro  1.771

In [46]:
#check last 5 records
consumer_price_fullfat_milk2L_monthly.tail()

Statistic    Month              Consumer Item  UNIT  VALUE
118  National Average Price  2021M10  Full fat milk per 2 litre  Euro  1.693
119  National Average Price  2021M11  Full fat milk per 2 litre  Euro  1.688
120  National Average Price  2021M12  Full fat milk per 2 litre  Euro  1.691
121  National Average Price  2022M01  Full fat milk per 2 litre  Euro  1.732
122  National Average Price  2022M02  Full fat milk per 2 litre  Euro  1.794

In [47]:
#check shape
consumer_price_fullfat_milk2L_monthly.shape

(123, 5)

In [48]:
#check data type
consumer_price_fullfat_milk2L_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 123 entries, 0 to 122
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Statistic      123 non-null    object 
 1   Month          123 non-null    object 
 2   Consumer Item  123 non-null    object 
 3   UNIT           123 non-null    object 
 4   VALUE          122 non-null    float64
dtypes: float64(1), object(4)
memory usage: 4.9+ KB


In [49]:
#check for nulls or missing values
consumer_price_fullfat_milk2L_monthly.isnull().sum()

Statistic        0
Month            0
Consumer Item    0
UNIT             0
VALUE            1
dtype: int64

Insights gained from above analysis for Consumer price fullfat milk data

1. Contains records from 2011 to 2022 with total 123 observations and 5 columns
2. Contains 1 null value in the value column
3. Month value is in object datatype and milk price is float datatype
4. unit is in euro per 2 litres and need to convert into 1 litre for like to like comparison

In [50]:
#check first 5 records
consumer_price_lowfat_milk1L_monthly.head()

Statistic    Month             Consumer Item  UNIT  VALUE
0  National Average Price  2011M12  Low fat milk per 1 litre  Euro    NaN
1  National Average Price  2012M01  Low fat milk per 1 litre  Euro  1.055
2  National Average Price  2012M02  Low fat milk per 1 litre  Euro  1.050
3  National Average Price  2012M03  Low fat milk per 1 litre  Euro  1.052
4  National Average Price  2012M04  Low fat milk per 1 litre  Euro  1.053

In [51]:
#check last 5 records
consumer_price_lowfat_milk1L_monthly.tail()

Statistic    Month             Consumer Item  UNIT  VALUE
118  National Average Price  2021M10  Low fat milk per 1 litre  Euro  1.027
119  National Average Price  2021M11  Low fat milk per 1 litre  Euro  1.026
120  National Average Price  2021M12  Low fat milk per 1 litre  Euro  1.020
121  National Average Price  2022M01  Low fat milk per 1 litre  Euro  1.049
122  National Average Price  2022M02  Low fat milk per 1 litre  Euro  1.077

In [52]:
#check shape
consumer_price_lowfat_milk1L_monthly.shape

(123, 5)

In [53]:
#check the data type
consumer_price_lowfat_milk1L_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 123 entries, 0 to 122
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Statistic      123 non-null    object 
 1   Month          123 non-null    object 
 2   Consumer Item  123 non-null    object 
 3   UNIT           123 non-null    object 
 4   VALUE          122 non-null    float64
dtypes: float64(1), object(4)
memory usage: 4.9+ KB


In [54]:
#check the null value or missing values
consumer_price_lowfat_milk1L_monthly.isnull().sum()

Statistic        0
Month            0
Consumer Item    0
UNIT             0
VALUE            1
dtype: int64

Insights gained from above analysis for Consumer price lowfat milk data

1. Contains records from 2011 to 2022 with total 123 observations and 5 columns
2. Contains 1 null value in the value column
3. Month value is in object datatype and milk price is float datatype
4. unit is in euro per 1 litre

In [55]:
#check first 5 records
cow_milk_intake_pasteury_creamery_monthly.head()

Statistic    Month  \
0  Intake of Cows Milk by Creameries and Pasteuri...  1975M01   
1  Intake of Cows Milk by Creameries and Pasteuri...  1975M02   
2  Intake of Cows Milk by Creameries and Pasteuri...  1975M03   
3  Intake of Cows Milk by Creameries and Pasteuri...  1975M04   
4  Intake of Cows Milk by Creameries and Pasteuri...  1975M05   

  Domestic or Import Source            UNIT  VALUE  
0                  Domestic  Million Litres   55.8  
1                  Domestic  Million Litres   84.4  
2                  Domestic  Million Litres  193.3  
3                  Domestic  Million Litres  327.5  
4                  Domestic  Million Litres  493.5

In [56]:
#check last 5 records
cow_milk_intake_pasteury_creamery_monthly.tail()

Statistic    Month  \
561  Intake of Cows Milk by Creameries and Pasteuri...  2021M10   
562  Intake of Cows Milk by Creameries and Pasteuri...  2021M11   
563  Intake of Cows Milk by Creameries and Pasteuri...  2021M12   
564  Intake of Cows Milk by Creameries and Pasteuri...  2022M01   
565  Intake of Cows Milk by Creameries and Pasteuri...  2022M02   

    Domestic or Import Source            UNIT  VALUE  
561                  Domestic  Million Litres  652.8  
562                  Domestic  Million Litres  460.6  
563                  Domestic  Million Litres  256.3  
564                  Domestic  Million Litres  175.7  
565                  Domestic  Million Litres  355.2

In [57]:
#check shape
cow_milk_intake_pasteury_creamery_monthly.shape

(566, 5)

In [58]:
#check data type
cow_milk_intake_pasteury_creamery_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 566 entries, 0 to 565
Data columns (total 5 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Statistic                  566 non-null    object 
 1   Month                      566 non-null    object 
 2   Domestic or Import Source  566 non-null    object 
 3   UNIT                       566 non-null    object 
 4   VALUE                      566 non-null    float64
dtypes: float64(1), object(4)
memory usage: 22.2+ KB


In [59]:
#check for null values or missing values
cow_milk_intake_pasteury_creamery_monthly.isnull().sum()

Statistic                    0
Month                        0
Domestic or Import Source    0
UNIT                         0
VALUE                        0
dtype: int64

Insights gained from above analysis for Cow milk pasteury creamey intake data

1. Contains records from 1975 to 2022 with total 566 observations and 5 columns
2. no null date in the data
3. Month value is in object datatype and intake milk value is float datatype
4. unit is in million litres

In [60]:
#check first 5 records
fertilizer_calciumammonium_nitrate_monthly.head()

Statistic    Month                  Type of Fertiliser  \
0  Fertiliser Price  1980M01  Calcium Ammonium Nitrate (27.5% N)   
1  Fertiliser Price  1980M02  Calcium Ammonium Nitrate (27.5% N)   
2  Fertiliser Price  1980M03  Calcium Ammonium Nitrate (27.5% N)   
3  Fertiliser Price  1980M04  Calcium Ammonium Nitrate (27.5% N)   
4  Fertiliser Price  1980M05  Calcium Ammonium Nitrate (27.5% N)   

             UNIT  VALUE  
0  Euro per Tonne    121  
1  Euro per Tonne    124  
2  Euro per Tonne    127  
3  Euro per Tonne    128  
4  Euro per Tonne    129

In [61]:
#check last 5 records
fertilizer_calciumammonium_nitrate_monthly.tail()

Statistic    Month                  Type of Fertiliser  \
501  Fertiliser Price  2021M10  Calcium Ammonium Nitrate (27.5% N)   
502  Fertiliser Price  2021M11  Calcium Ammonium Nitrate (27.5% N)   
503  Fertiliser Price  2021M12  Calcium Ammonium Nitrate (27.5% N)   
504  Fertiliser Price  2022M01  Calcium Ammonium Nitrate (27.5% N)   
505  Fertiliser Price  2022M02  Calcium Ammonium Nitrate (27.5% N)   

               UNIT  VALUE  
501  Euro per Tonne    384  
502  Euro per Tonne    395  
503  Euro per Tonne    499  
504  Euro per Tonne    637  
505  Euro per Tonne    697

In [62]:
#check shape
fertilizer_calciumammonium_nitrate_monthly.shape

(506, 5)

In [63]:
#check data type
fertilizer_calciumammonium_nitrate_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Statistic           506 non-null    object
 1   Month               506 non-null    object
 2   Type of Fertiliser  506 non-null    object
 3   UNIT                506 non-null    object
 4   VALUE               506 non-null    int64 
dtypes: int64(1), object(4)
memory usage: 19.9+ KB


In [64]:
#check for missing values or null values
fertilizer_calciumammonium_nitrate_monthly.isnull().sum()

Statistic             0
Month                 0
Type of Fertiliser    0
UNIT                  0
VALUE                 0
dtype: int64

Insights gained from above analysis for fertilizer calciumammonium nitrate data

1. Contains records from 1980 to 2022 with total 506 observations and 5 columns
2. no null date in the data
3. Month value is in object datatype and fertiliser price is integer datatype
3. unit is in euro per tonne

In [70]:
#check first 5 records
fertilizer_granularsuperphosphate_monthly.head()

Statistic    Month               Type of Fertiliser            UNIT  \
0  Fertiliser Price  1980M01  Granular Superphosphate (16% P)  Euro per Tonne   
1  Fertiliser Price  1980M02  Granular Superphosphate (16% P)  Euro per Tonne   
2  Fertiliser Price  1980M03  Granular Superphosphate (16% P)  Euro per Tonne   
3  Fertiliser Price  1980M04  Granular Superphosphate (16% P)  Euro per Tonne   
4  Fertiliser Price  1980M05  Granular Superphosphate (16% P)  Euro per Tonne   

  VALUE  
0   160  
1   162  
2   168  
3   168  
4   170

In [71]:
#check last 5 records
fertilizer_granularsuperphosphate_monthly.tail()

Statistic    Month               Type of Fertiliser  \
501  Fertiliser Price  2021M10  Granular Superphosphate (16% P)   
502  Fertiliser Price  2021M11  Granular Superphosphate (16% P)   
503  Fertiliser Price  2021M12  Granular Superphosphate (16% P)   
504  Fertiliser Price  2022M01  Granular Superphosphate (16% P)   
505  Fertiliser Price  2022M02  Granular Superphosphate (16% P)   

               UNIT VALUE  
501  Euro per Tonne     .  
502  Euro per Tonne     .  
503  Euro per Tonne     .  
504  Euro per Tonne     .  
505  Euro per Tonne     .

In [72]:
#check shape
fertilizer_granularsuperphosphate_monthly.shape

(506, 5)

In [73]:
#check data type
fertilizer_granularsuperphosphate_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Statistic           506 non-null    object
 1   Month               506 non-null    object
 2   Type of Fertiliser  506 non-null    object
 3   UNIT                506 non-null    object
 4   VALUE               429 non-null    object
dtypes: object(5)
memory usage: 19.9+ KB


In [74]:
#check missing values or null values
fertilizer_granularsuperphosphate_monthly.isnull().sum()

Statistic              0
Month                  0
Type of Fertiliser     0
UNIT                   0
VALUE                 77
dtype: int64

Insights gained from above analysis for fertilizer superphosphate data

1. Contains records from 1980 to 2022 with total 506 observations and 5 columns
2. Contains 77 null values in the fertiliser price data
3. Month value is in object datatype and fertiliser price is object datatype
4. unit is in euro per tonne

In [75]:
#check first 5 records
fertilizer_potash_muriate_monthly.head()

Statistic    Month         Type of Fertiliser            UNIT  VALUE
0  Fertiliser Price  1980M01  Muriate of Potash (50% K)  Euro per Tonne  109.0
1  Fertiliser Price  1980M02  Muriate of Potash (50% K)  Euro per Tonne  109.0
2  Fertiliser Price  1980M03  Muriate of Potash (50% K)  Euro per Tonne  114.0
3  Fertiliser Price  1980M04  Muriate of Potash (50% K)  Euro per Tonne  114.0
4  Fertiliser Price  1980M05  Muriate of Potash (50% K)  Euro per Tonne  114.0

In [76]:
#check last 5 records
fertilizer_potash_muriate_monthly.tail()

Statistic    Month         Type of Fertiliser            UNIT  \
501  Fertiliser Price  2021M10  Muriate of Potash (50% K)  Euro per Tonne   
502  Fertiliser Price  2021M11  Muriate of Potash (50% K)  Euro per Tonne   
503  Fertiliser Price  2021M12  Muriate of Potash (50% K)  Euro per Tonne   
504  Fertiliser Price  2022M01  Muriate of Potash (50% K)  Euro per Tonne   
505  Fertiliser Price  2022M02  Muriate of Potash (50% K)  Euro per Tonne   

     VALUE  
501  508.0  
502  509.0  
503  578.0  
504  668.0  
505  738.0

In [77]:
#check shape
fertilizer_potash_muriate_monthly.shape

(506, 5)

In [78]:
#check data type
fertilizer_potash_muriate_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Statistic           506 non-null    object 
 1   Month               506 non-null    object 
 2   Type of Fertiliser  506 non-null    object 
 3   UNIT                506 non-null    object 
 4   VALUE               465 non-null    float64
dtypes: float64(1), object(4)
memory usage: 19.9+ KB


In [79]:
#check for missing or null values
fertilizer_potash_muriate_monthly.isnull().sum()

Statistic              0
Month                  0
Type of Fertiliser     0
UNIT                   0
VALUE                 41
dtype: int64

Insights gained from above analysis for fertilizer potash muriate data

1. Contains records from 1980 to 2022 with total 506 observations and 5 columns
2. contains 41 null values in fertiliser price column
3. Month value is in object datatype and fertiliser price is float datatype
4. unit is in euro per tonne

In [80]:
#check first 5 records
fertilizer_urea_monthly.head()

Statistic    Month Type of Fertiliser            UNIT  VALUE
0  Fertiliser Price  1980M01       Urea (46% N)  Euro per Tonne    144
1  Fertiliser Price  1980M02       Urea (46% N)  Euro per Tonne    147
2  Fertiliser Price  1980M03       Urea (46% N)  Euro per Tonne    159
3  Fertiliser Price  1980M04       Urea (46% N)  Euro per Tonne    163
4  Fertiliser Price  1980M05       Urea (46% N)  Euro per Tonne    164

In [81]:
#check last 5 records
fertilizer_urea_monthly.tail()

Statistic    Month Type of Fertiliser            UNIT  VALUE
501  Fertiliser Price  2021M10       Urea (46% N)  Euro per Tonne    532
502  Fertiliser Price  2021M11       Urea (46% N)  Euro per Tonne    555
503  Fertiliser Price  2021M12       Urea (46% N)  Euro per Tonne    692
504  Fertiliser Price  2022M01       Urea (46% N)  Euro per Tonne    860
505  Fertiliser Price  2022M02       Urea (46% N)  Euro per Tonne    939

In [82]:
#check shape
fertilizer_urea_monthly.shape

(506, 5)

In [83]:
#check data type
fertilizer_urea_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Statistic           506 non-null    object
 1   Month               506 non-null    object
 2   Type of Fertiliser  506 non-null    object
 3   UNIT                506 non-null    object
 4   VALUE               506 non-null    int64 
dtypes: int64(1), object(4)
memory usage: 19.9+ KB


In [84]:
#check for missing or null values
fertilizer_urea_monthly.isnull().sum()

Statistic             0
Month                 0
Type of Fertiliser    0
UNIT                  0
VALUE                 0
dtype: int64

Insights gained from above analysis for fertilizer urea data

1. Contains records from 1980 to 2022 with total 506 observations and 5 columns
2. Contains no null values 
3. Month value is in object datatype and fertiliser price is integer datatype
4. unit is in euro per tonne

In [85]:
#check first 5 records
heifer_200_249kg_price_monthly.head()

Statistic    Month     Type of Cattle  UNIT  VALUE
0  Cattle Price per Head  1976M01  Heifers 200-249kg  Euro    NaN
1  Cattle Price per Head  1976M02  Heifers 200-249kg  Euro    NaN
2  Cattle Price per Head  1976M03  Heifers 200-249kg  Euro    NaN
3  Cattle Price per Head  1976M04  Heifers 200-249kg  Euro    NaN
4  Cattle Price per Head  1976M05  Heifers 200-249kg  Euro    NaN

In [86]:
#check last 5 records
heifer_200_249kg_price_monthly.tail()

Statistic    Month     Type of Cattle  UNIT   VALUE
549  Cattle Price per Head  2021M10  Heifers 200-249kg  Euro  516.00
550  Cattle Price per Head  2021M11  Heifers 200-249kg  Euro  419.66
551  Cattle Price per Head  2021M12  Heifers 200-249kg  Euro  490.37
552  Cattle Price per Head  2022M01  Heifers 200-249kg  Euro  553.25
553  Cattle Price per Head  2022M02  Heifers 200-249kg  Euro  587.61

In [87]:
#check shape
heifer_200_249kg_price_monthly.shape

(554, 5)

In [88]:
#check data type
heifer_200_249kg_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 554 entries, 0 to 553
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Statistic       554 non-null    object 
 1   Month           554 non-null    object 
 2   Type of Cattle  554 non-null    object 
 3   UNIT            554 non-null    object 
 4   VALUE           534 non-null    float64
dtypes: float64(1), object(4)
memory usage: 21.8+ KB


In [89]:
#check missing or null values
heifer_200_249kg_price_monthly.isnull().sum()

Statistic          0
Month              0
Type of Cattle     0
UNIT               0
VALUE             20
dtype: int64

Insights gained from above analysis for heifer200-249 data

1. Contains records from 1976 to 2022 with total 554 observations and 5 columns
2. Contains null values in heifer price column 
3. Month value is in object datatype and heifer price is float datatype
4. unit is in euro per head-count

In [90]:
#check first 5 records
heifer_250_299kg_price_monthly.head()

Statistic    Month     Type of Cattle  UNIT  VALUE
0  Cattle Price per Head  1976M01  Heifers 250-299kg  Euro    NaN
1  Cattle Price per Head  1976M02  Heifers 250-299kg  Euro    NaN
2  Cattle Price per Head  1976M03  Heifers 250-299kg  Euro    NaN
3  Cattle Price per Head  1976M04  Heifers 250-299kg  Euro    NaN
4  Cattle Price per Head  1976M05  Heifers 250-299kg  Euro    NaN

In [91]:
#check last 5 records
heifer_250_299kg_price_monthly.tail()

Statistic    Month     Type of Cattle  UNIT   VALUE
549  Cattle Price per Head  2021M10  Heifers 250-299kg  Euro  529.67
550  Cattle Price per Head  2021M11  Heifers 250-299kg  Euro  547.42
551  Cattle Price per Head  2021M12  Heifers 250-299kg  Euro  623.57
552  Cattle Price per Head  2022M01  Heifers 250-299kg  Euro  640.07
553  Cattle Price per Head  2022M02  Heifers 250-299kg  Euro  668.57

In [92]:
#check shape
heifer_250_299kg_price_monthly.shape

(554, 5)

In [93]:
#check data type
heifer_250_299kg_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 554 entries, 0 to 553
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Statistic       554 non-null    object 
 1   Month           554 non-null    object 
 2   Type of Cattle  554 non-null    object 
 3   UNIT            554 non-null    object 
 4   VALUE           535 non-null    float64
dtypes: float64(1), object(4)
memory usage: 21.8+ KB


In [94]:
#check for null values or missing values
heifer_250_299kg_price_monthly.isnull().sum()

Statistic          0
Month              0
Type of Cattle     0
UNIT               0
VALUE             19
dtype: int64

Insights gained from above analysis for heifer250-299 data

1. Contains records from 1976 to 2022 with total 554 observations and 5 columns
2. Contains null values in heifer price column
3. Month value is in object datatype and heifer price is float datatype
4. unit is in euro per head-count

In [95]:
#check for first 5 records
heifer_300_349kg_price_monthly.head()

Statistic    Month     Type of Cattle  UNIT   VALUE
0  Cattle Price per Head  1976M01  Heifers 300-349kg  Euro  220.20
1  Cattle Price per Head  1976M02  Heifers 300-349kg  Euro  215.39
2  Cattle Price per Head  1976M03  Heifers 300-349kg  Euro  236.20
3  Cattle Price per Head  1976M04  Heifers 300-349kg  Euro  239.51
4  Cattle Price per Head  1976M05  Heifers 300-349kg  Euro  260.88

In [96]:
#check for last 5 records
heifer_300_349kg_price_monthly.tail()

Statistic    Month     Type of Cattle  UNIT   VALUE
549  Cattle Price per Head  2021M10  Heifers 300-349kg  Euro  656.53
550  Cattle Price per Head  2021M11  Heifers 300-349kg  Euro  641.39
551  Cattle Price per Head  2021M12  Heifers 300-349kg  Euro  693.47
552  Cattle Price per Head  2022M01  Heifers 300-349kg  Euro  765.68
553  Cattle Price per Head  2022M02  Heifers 300-349kg  Euro  776.96

In [97]:
#check shape
heifer_300_349kg_price_monthly.shape

(554, 5)

In [98]:
#check data type
heifer_300_349kg_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 554 entries, 0 to 553
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Statistic       554 non-null    object 
 1   Month           554 non-null    object 
 2   Type of Cattle  554 non-null    object 
 3   UNIT            554 non-null    object 
 4   VALUE           550 non-null    float64
dtypes: float64(1), object(4)
memory usage: 21.8+ KB


In [99]:
#check for null values or missing values
heifer_300_349kg_price_monthly.isnull().sum()

Statistic         0
Month             0
Type of Cattle    0
UNIT              0
VALUE             4
dtype: int64

Insights gained from above analysis for heifer300-349 data

1. Contains records from 1976 to 2022 with total 554 observations and 5 columns
2. Contains null values in heifer price column
3. Month value is in object datatype and heifer price is float datatype
4. unit is in euro per head-count

In [100]:
#check for first 5 records
heifer_350_399kg_price_monthly.head()

Statistic    Month     Type of Cattle  UNIT   VALUE
0  Cattle Price per Head  1976M01  Heifers 350-399kg  Euro  252.07
1  Cattle Price per Head  1976M02  Heifers 350-399kg  Euro  276.93
2  Cattle Price per Head  1976M03  Heifers 350-399kg  Euro  277.09
3  Cattle Price per Head  1976M04  Heifers 350-399kg  Euro  297.18
4  Cattle Price per Head  1976M05  Heifers 350-399kg  Euro  314.56

In [101]:
#check for last 5 records
heifer_350_399kg_price_monthly.tail()

Statistic    Month     Type of Cattle  UNIT   VALUE
549  Cattle Price per Head  2021M10  Heifers 350-399kg  Euro  777.37
550  Cattle Price per Head  2021M11  Heifers 350-399kg  Euro  762.83
551  Cattle Price per Head  2021M12  Heifers 350-399kg  Euro  806.00
552  Cattle Price per Head  2022M01  Heifers 350-399kg  Euro  857.22
553  Cattle Price per Head  2022M02  Heifers 350-399kg  Euro  910.01

In [102]:
#check shape
heifer_350_399kg_price_monthly.shape

(554, 5)

In [103]:
#check data type
heifer_350_399kg_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 554 entries, 0 to 553
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Statistic       554 non-null    object 
 1   Month           554 non-null    object 
 2   Type of Cattle  554 non-null    object 
 3   UNIT            554 non-null    object 
 4   VALUE           550 non-null    float64
dtypes: float64(1), object(4)
memory usage: 21.8+ KB


In [104]:
#check the null values or missing values
heifer_350_399kg_price_monthly.isnull().sum()

Statistic         0
Month             0
Type of Cattle    0
UNIT              0
VALUE             4
dtype: int64

Insights gained from above analysis for heifer350-399 data

1. Contains records from 1976 to 2022 with total 554 observations and 5 columns
2. Contains null values in heifer price column
3. Month value is in object datatype and heifer price is float datatype
4. unit is in euro per head-count

In [105]:
#check first 5 records
heifer_400_449kg_price_monthly.head()

Statistic    Month     Type of Cattle  UNIT   VALUE
0  Cattle Price per Head  1976M01  Heifers 400-449kg  Euro  309.22
1  Cattle Price per Head  1976M02  Heifers 400-449kg  Euro  309.89
2  Cattle Price per Head  1976M03  Heifers 400-449kg  Euro  333.41
3  Cattle Price per Head  1976M04  Heifers 400-449kg  Euro  353.28
4  Cattle Price per Head  1976M05  Heifers 400-449kg  Euro  351.32

In [106]:
#check last 5 records
heifer_400_449kg_price_monthly.tail()

Statistic    Month     Type of Cattle  UNIT    VALUE
549  Cattle Price per Head  2021M10  Heifers 400-449kg  Euro   890.95
550  Cattle Price per Head  2021M11  Heifers 400-449kg  Euro   888.34
551  Cattle Price per Head  2021M12  Heifers 400-449kg  Euro   912.35
552  Cattle Price per Head  2022M01  Heifers 400-449kg  Euro   970.51
553  Cattle Price per Head  2022M02  Heifers 400-449kg  Euro  1034.15

In [107]:
#check shape
heifer_400_449kg_price_monthly.shape

(554, 5)

In [108]:
#check data type
heifer_400_449kg_price_monthly.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 554 entries, 0 to 553
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Statistic       554 non-null    object 
 1   Month           554 non-null    object 
 2   Type of Cattle  554 non-null    object 
 3   UNIT            554 non-null    object 
 4   VALUE           550 non-null    float64
dtypes: float64(1), object(4)
memory usage: 21.8+ KB


In [109]:
#check missing values or null values
heifer_400_449kg_price_monthly.isnull().sum()

Statistic         0
Month             0
Type of Cattle    0
UNIT              0
VALUE             4
dtype: int64

Insights gained from above analysis for heifer400-449 data

1. Contains records from 1976 to 2022 with total 554 observations and 5 columns
2. Contains null values in heifer price column
3. Month value is in object datatype and heifer price is float datatype
4. unit is in euro per head-count